engine/src/org/pentaho/di/cluster/SlaveServer.java

```diff
@@ -139,49 +142,52 @@ public static int getBackoffIncrements() {
 
   private LogChannelInterface log;
 
-  private String name;
+  private volatile String name;
```

Comment:
Note - I believe if you're using a ReentrantReadWriteLock, and you're always locking before reading and writing, you *shouldn't* use the volatile keyword. The volatile keyword should only be used when accessing a variable _outside_ the lock.